spec/GraphQLOverHTTP.md

```diff
@@ -511,19 +511,16 @@ execution regardless of validation errors.
 
 ## Status Codes
 
-In case of errors that completely prevent the generation of a well-formed
-_GraphQL response_, the server SHOULD respond with the appropriate status code
-depending on the concrete error condition, and MUST NOT respond with a `2xx`
-status code when using the `application/graphql-response+json` media type.
+The status codes depends on the media type with which the GraphQL response will
+be served.
 
-Note: Typically the appropriate status code will be `400` (Bad Request).
-
-Note: This rule is "should" to maintain compatibility with legacy servers which
-can return 200 status codes even when this type of error occurs, but only when
-not using the `application/graphql-response+json` media type.
-
-Otherwise, the status codes depends on the media type with which the GraphQL
-response will be served:
+Note: when using the `application/graphql-response+json` media type, the status
+code is primarily aimed for intermediary servers that do not understand GraphQL.
+In these cases, clients can rely on the response being a well-formed _GraphQL
+response_ regardless of the status code. When using the `application/json` media
+type on the other hand, clients may use a `2xx` status code as an indication
+that the body contains a well-formed _GraphQL response_. See
+[processing a response](#sec-Processing-a-response) for more details.
```

Comment:
This seems confusing to me.

> Note: when using the `application/graphql-response+json` media type

To me this implies that the response is the above type

> the status code is primarily aimed for intermediary servers that do not understand GraphQL

In which case the status code was provided by the GraphQL server.

But I am guessing the intent here is more like "when expecting the `application/graphql-response+json` media type", then this and that.

On a side note, I would say the status code is always "for" the client, not "for" intermediary servers.

Reply:
> To me this implies that the response is the above type

👍  I'll reword

> I would say the status code is always "for" the client, not "for" intermediary servers.

This has been my expectation for the last 7 years 😅. I agree that status code in general are meant for clients but the more I look at the `graphql-response+json` new way to do things, the more I feel I have been wrong all this time... (See also [Discord conversation](https://discord.com/channels/625400653321076807/863141924126588958/1266313412585979955)). 

If I'm a client and I'm receiving a `graphql-response+json` response, I'll probably not even look at the status code and just forward everything to the user.

Reply:
I have reworded this to remove the "primarily aimed". We don't know how people are going to use it so I went for something more neutral.